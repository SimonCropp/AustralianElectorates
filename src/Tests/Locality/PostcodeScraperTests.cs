using System.IO;
using System.Linq;
using System.Threading.Tasks;
using Xunit;
using Xunit.Abstractions;

public class PostcodeScraperTests :
    XunitLoggingBase
{
    [Fact]
    [Trait("Category", "Integration")]
    public async Task Run()
    {
        var data = await PostcodeScraper.Run();
        File.Delete(DataLocations.LocalitiesPath);
        JsonSerializer.Serialize(data, DataLocations.LocalitiesPath);
        ObjectApprover.Verify(data.Take(10));
    }

    [Fact]
    public async Task Specific()
    {
        ObjectApprover.Verify(await PostcodeScraper.GetAECDataForPostcode("2612"));
    }

    public PostcodeScraperTests(ITestOutputHelper output) :
        base(output)
    {
    }
}