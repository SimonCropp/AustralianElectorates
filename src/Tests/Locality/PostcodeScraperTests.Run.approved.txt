[
  {
    State: 'NT',
    LocalityOrSuburb: 'DARWIN',
    Postcode: 800,
    Electorate: 'SOLOMON'
  },
  {
    State: 'NT',
    LocalityOrSuburb: 'NGARLU NGARLU',
    Postcode: 801,
    Electorate: 'LINGIARI'
  },
  {
    State: 'NT',
    LocalityOrSuburb: 'SANDY BAY',
    Postcode: 801,
    Electorate: 'LINGIARI'
  },
  {
    State: 'NT',
    LocalityOrSuburb: 'ALAWA',
    Postcode: 810,
    Electorate: 'SOLOMON'
  },
  {
    State: 'NT',
    LocalityOrSuburb: 'BRINKIN',
    Postcode: 810,
    Electorate: 'SOLOMON'
  },
  {
    State: 'NT',
    LocalityOrSuburb: 'BUFFALO CREEK',
    Postcode: 810,
    Electorate: 'SOLOMON'
  },
  {
    State: 'NT',
    LocalityOrSuburb: 'CASUARINA',
    Postcode: 810,
    Electorate: 'SOLOMON'
  },
  {
    State: 'NT',
    LocalityOrSuburb: 'COCONUT GROVE',
    Postcode: 810,
    Electorate: 'SOLOMON'
  },
  {
    State: 'NT',
    LocalityOrSuburb: 'JINGILI',
    Postcode: 810,
    Electorate: 'SOLOMON'
  },
  {
    State: 'NT',
    LocalityOrSuburb: 'KULALUK',
    Postcode: 810,
    Electorate: 'SOLOMON'
  }
]